packages/dart_code_builder/src/example.candy

```diff
@@ -0,0 +1,140 @@
+  list.append(Function(
+    "main",
+    annotations = List.of1<Annotation>(
+      Annotation(
+        (UserType(
+          "Cached",
+          parameters = List.empty<DartType>(),
+          prefix = None<String>(),
+        ) as Expression).call(
+          positionalArguments = List.empty<Expression>(),
+          namedArguments = Map.empty<String, Expression>(),
+          typeArguments = List.empty<DartType>(),
+        )
+      )
+    ),
+    _returns = Some<DartType>(UserType(
+      "Foo",
+      parameters = List.empty<DartType>(),
+      prefix = Some<String>("foo"),
+    )),
+    requiredParameters = List.of1<Parameter>(
+      Parameter(
+        "args",
+        type = Some<DartType>(UserType(
+          "List",
+          parameters = List.of1<DartType>(UserType(
+            "String",
+            parameters = List.empty<DartType>(),
+            prefix = None<String>(),
+          )),
+          prefix = None<String>(),
+        )),
+        defaultValue = None<Expression>(),
+      ),
+    ),
+    positionalParameters = List.of1<Parameter>(
+      Parameter(
+        "number",
+        type = Some<DartType>(UserType("int", List.empty<DartType>(), None<String>())),
+        default = Some<Expression>(Literal(123)),
+      ),
+    ),
+    namedParameters = List.empty<Parameter>(),
+    body = None<Body>(),
+  ))
+  list.append(Class(
+    "Foo",
+    annotations = List.of1<Annotation>(Annotation(Identifier("data", prefix = None<String>()))),
+    extends_ = Some<DartType>(UserType("Bar", List.empty<DartType>(), None<String>())),
+    implements_ = List.of3<DartType>(
+      UserType("Baz", List.empty<DartType>(), None<String>()),
+      UserType("Wheez", List.empty<DartType>(), None<String>()),
+      UserType("Fnop", List.empty<DartType>(), None<String>()),
+    ),
+    with_ = List.of2<DartType>(
+      UserType("Whup", List.empty<DartType>(), None<String>()),
+      UserType("Flang", List.empty<DartType>(), None<String>()),
+    ),
+    body = List.of4<Field | Getter | Setter | Function>(
+      // static const numBlubs = 123;
+      Field("numBlubs", isStatic = true, mutability = Const(), Some<Expression>(Literal(123))),
+      // bool get isEmpty => this.length == 0;
+      Getter(
+        "isEmpty",
+        type = UserType("bool", List.empty<DartType>(), None<String>()),
+        body = Some<Body>(InlineBody(this_.dot("length").equals(Literal(0)))),
+      ),
+      // set isEmpty(bool value) {
+      //   if (value) {
+      //     length = 0;
+      //   } else {
+      //     add(1);
+      //   }
+      // }
+      Setter(
+        "isEmpty",
+        parameter = Parameter(
+          "value",
+          type = Some<DartType>(UserType("bool", List.empty<DartType>(), None<String>())),
+          defaultValue = None<Expression>(),
+        ),
+        body = Some<Body>(BlockBody(List.of1<Statement>(
+          If(
+            Identifier("value", None<String>()),
+            then = List.of1<Statement>(
```

Comment:
In the Dart analyzer, this is a single statement which can be a `Block`.

* https://pub.dev/documentation/analyzer/latest/dart_ast_ast/IfStatement-class.html
* https://pub.dev/documentation/analyzer/latest/dart_ast_ast/Statement-class.html
* https://pub.dev/documentation/analyzer/latest/dart_ast_ast/Block-class.html